### PR TITLE
gammu-smsd service

### PR DESCRIPTION
**Pull request by zohl**
This patch adds a service built on top of gammu utility.
Basically, it stores recieved SMS and take care of interaction with the device.

What works:
- File system backend (storing sms in a plain text)
- Postgresql backend

What doesn't works:
- sqlite and any other DBI-based connections.
  I dig a little into the code and found out that the daemon on `dbi_initialize` function. It seems that something related to driver path is went wrong.

What wasn't implemented:
- ODBC backend
- MySQL backend

All options except `user` and `group` are described in the official manpages (gammurc and gammu-smsdrc). (These two options are here for security reasons.)

### CODE REVIEW

File: nixos/modules/services/misc/gammu-smsd.nix [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+            description = "Where SMSes with error in transmission is placed";
+          };
+        };
+
+        sql = {
+          driver = mkOption {
+            type = types.string;
+            default = "";
+            description = "BD driver to use";
+            example = [ "native_mysql" "native_pgsql" "odbc" "dbi" ];
```
Comment (bjornfor) [inline nixos/modules/services/misc/gammu-smsd.nix]: The type says string, but the example says list of strings. Which one is it? Can more than one database driver be enabled at once? If not, maybe use "enum" type instead. (Btw, types.string is deprecated, use types.str instead.) Also, BD => DB typo :in description -)

**Comment by zohl** [comment]
@abbradar, @bjornfor, thank you for the comments. I've fixed the commits according to your recommendations and updated the PR.

**Comment by zohl** [comment]
@abbradar, fixed and updated

**Comment by abbradar** [comment]
Some more comments. Thanks you very much for bearing with me and sorry for not noticing this earlier!

**Comment by zohl** [comment]
Some more updates. That's ok, live and learn! :smiley:

**Comment by zohl** [comment]
Updated

**Comment by abbradar** [comment]
Also, the commit name feels somewhat invalid -- _service_ doesn't have a version, _package_ does -- we usually name them just `somed service: init`.

**Comment by abbradar** [comment]
Thanks! This looks good to me now, and don't sweat much over the commit name, I think -- I've bugged you enough already ~_^. I'll wait for several days for others to be able to express their thoughs and merge. Thank you for your work again!

**Comment by abbradar** [comment]
Notice that I've changed service UID/GID -- those were already taken.

**Pull request merged**